File: cpp/1582.cpp
using std::count;
class Solution {
  public:
    int numSpecial(vector<vector<int>>& mat) {
      int rows = mat.size();
      int cols = mat[0].size();

      vector<int> rowCount(rows);
      vector<int> colCount(cols);

      int result = 0;

      for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
          if (mat[i][j]) {
            rowCount[i]++;
            colCount[j]++;
          }
        }
      }

      for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
          if (mat[i][j] && rowCount[i] == 1 && colCount[j] == 1) {
            result++;
          }
        }
      }

      return result;
    }
};

File: go/1582.go
func numSpecial(mat [][]int) int {

  rows := len(mat)
  cols := len(mat[0])

  rowCount := make([]int, rows)
  colCount := make([]int, cols)

  result := 0

  for i:=0; i < rows; i++ {
      for k:=0; k < cols; k++ {
          if mat[i][k] == 1 {
              rowCount[i] += 1
              colCount[k] += 1
          }
      }
  }

  for i:=0; i < rows; i++ {
      for k:=0; k < cols; k++ {
          if mat[i][k] == 1 && rowCount[i] == 1 && colCount[k] == 1 {
              result += 1
          }
      }
  }

  return result
}
